File: OCTOBER/KINHYEONJI/BOJ1788.cpp
#include <iostream>

using namespace std;

long long arr[1'000'002];

int main() {
    arr[0] = 0;
    arr[1] = 1;
    for (int i = 2; i < 1000001; i++) {
        arr[i] = (arr[i - 1] + arr[i - 2]) % 1'000'000'000;
    }
    int num;
    cin >> num;
    if (num > 0) {
        cout << 1 << '\n' << arr[num];

    } else if (num < 0) {
        if (abs(num) % 2 == 0) {
            cout << -1 << '\n' << arr[abs(num)];
        } else {
            cout << 1 << '\n' << arr[abs(num)];
        }
    } else {
        cout << 0 << '\n' << 0;
    }
}

File: OCTOBER/GISEONOH/BOJ1788.py
N = int(input())
arr = [0, 1]

for i in range(2, abs(N) + 1):
    arr.append((arr[i-1] + arr[i-2]) % 1000000000)

if N < 0:
    if abs(N) % 2 == 0:
        print(-1)
    else:
        print(1)
elif N > 0:
    print(1)
else:
    print(0)

print(arr[abs(N)])